plasmapy/tests/helpers/tests/test_expected.py

```diff
@@ -0,0 +1,178 @@
+    Case(expected_warning_and_value, "expected_outcome", expected_warning_and_value),
+]
+
+
+@pytest.mark.parametrize("case", expected_outcome_test_cases)
+def test_expected_test_outcome_attributes(case: Case):
+    """Test that `ExpectedTestOutcome` attributes each return the expected values."""
+    expected_outcome = ExpectedTestOutcome(case.argument)
+    result = expected_outcome.__getattribute__(case.attribute)
+    if result is not case.correct_outcome and result != case.correct_outcome:
+        pytest.fail(
+            f"ExpectedTestOutcome({case.argument}) results in {repr(result)} "
+            f"but should result in {case.correct_outcome}."
+        )
+
+
+exception_to_be_raised = RuntimeError
+
+exception_raising_cases = [
+    Case(expected_exception, "expected_warning", exception_to_be_raised),
+    Case(expected_exception, "expected_value", exception_to_be_raised),
+    Case(expected_warning, "expected_exception", exception_to_be_raised),
+    Case(expected_warning, "expected_value", exception_to_be_raised),
+    Case(expected_value, "expected_exception", exception_to_be_raised),
+    Case(expected_value, "expected_warning", exception_to_be_raised),
+    Case(expected_value_and_warning, "expected_exception", exception_to_be_raised),
+    Case(expected_warning_and_value, "expected_exception", exception_to_be_raised),
+]
+
+
+@pytest.mark.parametrize("case", exception_raising_cases)
+def test_expected_test_outcome_exceptions(case: Case):
+    """Test that attributes of `ExpectedTestOutcome` raise exceptions as needed."""
+    if not issubclass(case.correct_outcome, Exception):
+        raise TypeError(
+            "Incorrect test setup: the expected outcome must be an exception."
+        )
+    with pytest.raises(case.correct_outcome):
+        expected_outcome = ExpectedTestOutcome(case.argument)
+        result = expected_outcome.__getattribute__(case.attribute)
+        pytest.fail(
+            f"The ExpectedTestOutcome instance for {case.argument} did not "
+            f"raise the expected exception but instead returned {result}."
+        )
+
+
+is_warning_test_inputs = [
+    (Warning, True),
+    (UserWarning, True),
+    (PlasmaPyWarning, True),
+    (Exception, False),
+    (BaseException, False),
+    ("", False),
+    ((Warning, 1), False),
+]
+
+
+@pytest.mark.parametrize("possible_warning, actually_a_warning", is_warning_test_inputs)
+def test__is_warning(possible_warning, actually_a_warning: bool):
+    """
+    Test that `~plasmapy.utils.pytest_helpers.expected._is_warning`
+    returns `True` for warnings and `False` for other objects.
+    """
+    interpreted_as_warning = _is_warning(possible_warning)
+    if interpreted_as_warning != actually_a_warning:
+        pytest.fail(
+            f"_is_warning({repr(possible_warning)}) should return "
+            f"{actually_a_warning}, but is instead returning "
+            f"{interpreted_as_warning}."
+        )
```

Comment:
```suggestion
    assert interpreted_as_warning == actually_a_warning
```

It's also usually simpler to think in terms of true statements rather than their negations.